### PR TITLE
Imran Turganov, HW1, All done

### PR DESCRIPTION
**Pull request by 1MRUN**

### CODE REVIEW

**Comment by kajigor** [comment]
Put your name into the PR title, please.

**Comment by 1MRUN** [comment]
> Put your name into the PR title, please.

done

File: Main.hs [hunk 1]
```diff
@@ -4,22 +4,46 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short (h : x : y : t) = False
```
Comment (kajigor) [inline Main.hs:7]: Use wildcards instead of introducing variable names if you don't use them.

File: Main.hs [hunk 2]
```diff
@@ -4,22 +4,46 @@ import Control.Monad (unless)
...
+ageOn planet ageInSeconds 
+    | planet == "Mercury" = ageInSeconds / (0.2408467 * 31557600)
+    | planet == "Venus" = ageInSeconds / (0.61519726 * 31557600)
+    | planet == "Earth" = ageInSeconds / (1  * 31557600)
+    | planet == "Mars" = ageInSeconds / (1.8808158  * 31557600)
+    | planet == "Jupiter" = ageInSeconds / (11.862615  * 31557600)
+    | planet == "Saturn" = ageInSeconds / (29.447498 * 31557600)
+    | planet == "Uranus" = ageInSeconds / (84.016846 * 31557600)
+    | planet == "Neptune" = ageInSeconds / (164.79132 * 31557600)
+    | otherwise = error "no planet"
```
Comment (kajigor) [inline Main.hs:38]: You have a lot of code duplication here. You can introduce a function that does the necessary computations and call it instead.

**Review by kajigor — CHANGES_REQUESTED** [review]

File: Main.hs (hunk 2 above)
Comment (1MRUN) [reply Main.hs]: Please check if I changed it right.

File: Main.hs [hunk 3]
```diff
@@ -4,22 +4,48 @@ import Control.Monad (unless)
...
+    | planet == "Venus" = time(ageInSeconds,0.61519726)
+    | planet == "Earth" = time(ageInSeconds,1)
+    | planet == "Mars" = time(ageInSeconds,1.8808158)
+    | planet == "Jupiter" = time(ageInSeconds,11.862615)
+    | planet == "Saturn" = time(ageInSeconds,29.447498)
+    | planet == "Uranus" = time(ageInSeconds,84.016846)
+    | planet == "Neptune" = time(ageInSeconds,164.79132)
+    | otherwise = error "no planet"
+    where 
+        time (sec,frac) = sec / (frac * 31557600)
```
Comment (kajigor) [inline Main.hs:40]: No need to wrap arguments into a tuple. We'll see why today.

File: Main.hs [hunk 4]
```diff
@@ -4,22 +4,48 @@ import Control.Monad (unless)
...
 
 fizzBuzz :: [String]
-fizzBuzz = undefined 
+fizzBuzz = [logic x | x <- [1..]]
+    where
+        logic x  
+            | x `mod` 15 == 0 = "FizzBuzz"
+            | x `mod` 5 == 0 = "Buzz"
+            | x `mod` 3 == 0 = "Fizz"
+            | otherwise = show(x)
```
Comment (kajigor) [inline Main.hs:26]: Avoid using unnecessary parentheses: Haskell is not Java, you don't need to put arguments of a function in parentheses.

**Comment by kajigor** [comment]
ok.

**Pull request closed without merging**